text/3-entrees/fixed-point.md

```diff
@@ -0,0 +1,329 @@
+
+As in the `Permill` example, we begin by declaring the storage item. With substrate-fixed, there is not a `one` function, but there is a `from_num` function that we use to set the storage item's default value. This `from_num` method and it's counterpart `to-num` are your primary ways of converting between substrate-fixed types and Rust primitive types. If your use case does a lot of fixed-point arithmetic, like ours does, it is advisable to keep your data in substrate-fixed types.
+
+> We're able to use `U16F16` as a storage item type because it, and the other substrate-fixed types, implement the parity scale codec.
+
+```rust, ignore
+decl_storage! {
+	trait Store for Module<T: Trait> as Example {
+		// --snip--
+
+		/// Substrate-fixed accumulator, value starts at 1 (multiplicative identity)
+		FixedAccumulator get(fn fixed_value): U16F16 = U16F16::from_num(1);
+	}
+}
+```
+
+Next we implement the extrinsic that allows users to update the accumulator by multiplying in a new value.
+
+```rust, ignore
+fn update_fixed(origin, new_factor: U16F16) -> DispatchResult {
+	ensure_signed(origin)?;
+
+	let old_accumulated = Self::fixed_value();
+
+	// Multiply, handling overflow
+	let new_product = old_accumulated.checked_mul(new_factor)
+		.ok_or(Error::<T>::Overflow)?;
+
+	// Write the new value to storage
+	FixedAccumulator::put(new_product);
+
+	// Emit event
+	Self::deposit_event(Event::FixedUpdated(new_factor, new_product));
+	Ok(())
+}
+```
+
+This extrinsic is quite similar to the `Permill` version with one notable difference. Because `U16F16` handles numbers greater than one, overflow is possible, and we need to handle it. The error handling here is straightforward, the important part is just that you remember to do it.
+
+This example has shown the fundamentals of substrate-fixed, but this library has much more to offer as we'll see in the compounding interest example.
+
+### Manual Accumulator
+
+In this final accumulator implementation, we manually track fixed point numbers using Rust's native `u32` as the underlying data type. This example is educational, but is only practical in the simplest scenarios. Generally you will have a ~~more fun~~ less error-prone time coding if you use one of the previous two fixed-point types in your real-world applications.
+
+Fixed point is not very complex conceptually. We represent fractional numbers as regular old integers, and we decide in advance to consider some of the place values fractional. It's just like saying we'll omit the decimal point when talking about money and all agree that "1995" actually _means_ €19.95. This is exactly how Substrate's [Balances pallet](https://substrate.dev/rustdocs/master/pallet_balances/index.html) works, a tradition that's been in blockchain since Bitcon. In our example we will treat 16 bits as integer values, and 16 as fractional, just as substrate-fixed's `U16F16` did.
```

Comment:
Not 100% sure on this, but I think only US puts the $ before the number. 
```suggestion
Fixed point is not very complex conceptually. We represent fractional numbers as regular old integers, and we decide in advance to consider some of the place values fractional. It's just like saying we'll omit the decimal point when talking about money and all agree that "1995" actually _means_ 19.95 €. This is exactly how Substrate's [Balances pallet](https://substrate.dev/rustdocs/master/pallet_balances/index.html) works, a tradition that's been in blockchain since Bitcon. In our example we will treat 16 bits as integer values, and 16 as fractional, just as substrate-fixed's `U16F16` did.
```